BangazonAPI/Controllers/CustomerController.cs

```diff
@@ -0,0 +1,214 @@
+                    customer.Id = newId;
+                    return CreatedAtRoute("GetCustomer", new
+                    {
+                        id = newId
+                    }, customer);
+                }
+            }
+
+        }
+        [HttpPut("{id}")]
+        public async Task<IActionResult> Put([FromRoute] int id, [FromBody] Customer customer)
+        {
+            try
+            {
+
+                using (SqlConnection conn = Connection)
+                {
+                    conn.Open();
+                    using (SqlCommand cmd = conn.CreateCommand())
+                    {
+                        cmd.CommandText = @"UPDATE Customer set
+                            FirstName = @FirstName ,
+                            LastName = @LastName
+                            WHERE Id = @id";
+                        cmd.Parameters.Add(new SqlParameter("@FirstName", customer.FirstName));
+                        cmd.Parameters.Add(new SqlParameter("@LastName", customer.LastName));
+                        cmd.Parameters.Add(new SqlParameter("@Id", id));
+
+                        int rowsAffected = cmd.ExecuteNonQuery();
+                        if (rowsAffected > 0)
+                        {
+                            return new StatusCodeResult(StatusCodes.Status204NoContent);
+                        }
+                        throw new Exception("No rows affected");
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                if (!CustomerExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+        }
+
+        [HttpGet("{id}", Name = "GetCustomer")]
+        public async Task<IActionResult> Get([FromRoute] int id)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = "SELECT Id, FirstName, LastName  FROM Customer WHERE Id = @id ";
+                    cmd.Parameters.Add(new SqlParameter("@id", id));
+                    SqlDataReader reader = cmd.ExecuteReader();
+                    Customer customer = null;
+                    if (reader.Read())
+                    {
+                        customer = new Customer
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                            FirstName = reader.GetString(reader.GetOrdinal("FirstName")),
+                            LastName = reader.GetString(reader.GetOrdinal("LastName")),
+                        };
+                    }
+                    reader.Close();
+
+                    return Ok(customer);
+                }
+            }
+        }
+
+
+        [HttpGet]
+        public IEnumerable<Customer> Get(string q)
```

Comment:
I would like to see
`public IEnumerable<Customer> Get(string q)`
and
`public async Task<IActionResult> Get()`
combined into one method that has an optional `q` query string parameter.

Reply:
thats more of a refactoring step atm we are just trying to finish this sprint

Reply:
Please combine them into a single method.

Reply:
why its unnecessary work that will keep us from hitting the end of our sprint successfully,  mind explaining your rationale?

Reply:
fixed